c2rust-analyze/src/context.rs

```diff
@@ -182,6 +193,27 @@ impl<'tcx> GlobalAnalysisCtxt<'tcx> {
             self.assign_pointer_to_field(field);
         }
     }
+
+    pub fn fn_failed(&mut self, did: DefId) -> bool {
+        self.fns_failed.contains_key(&did)
+    }
+
+    pub fn mark_fn_failed(&mut self, did: DefId, detail: PanicDetail) {
```

Comment:
I feel like I must be missing something; where is this called on any failing code? I only see the recursive call below